### PR TITLE
Added printer name to log entry and added filter to ignore classes in cups

### PR DESCRIPTION
**Pull request by tokarlss**
Hello!

Please consider including this patch in your code.

I discovered that we had problems in our environment that the cups-connector failed to download some PPD files. I initially added print out of printer name in the getPPD function to be able to diagnose the problem (one of the commits). I discovered that some of our printers that are defined as classes failed to present a PPD file and none of the classes are really useful as cloudprint printers. So i added a function to filter out classes the same way the raw printers can be filtered including the configuration for it.
Doing this i discovered that the configuration to enable/disable ignoring raw printers was not passed to cups.go hence the raw printers was always ignored, i have added passing both the config to ignore raw printers and the new option to ignore classes.

I hope you find this useful and can consider adding this to your source. 

//Tomas Karlsson (tokarlss)

### CODE REVIEW

File: cups/core.go [hunk 1]
```diff
@@ -191,8 +191,8 @@ func (cc *cupsCore) getPPD(printername *C.char, modtime *C.time_t) (*C.char, err
 		C.free(unsafe.Pointer(buffer))
 		cupsLastError := C.cupsLastError()
 		if cupsLastError != C.IPP_STATUS_OK {
-			return nil, fmt.Errorf("Failed to call cupsGetPPD3(): %d %s",
-				int(cupsLastError), C.GoString(C.cupsLastErrorString()))
+			return nil, fmt.Errorf("Failed to call cupsGetPPD3() for printer %s: %d %s",
```
Comment (jacobmarble) [inline cups/core.go]: Undo this change, then go to file cups.go, func addPPDDescriptionToPrinters. Change log.Error(err) to log.ErrorPrinter(p.Name, err).

File: lib/config.go [hunk 2]
```diff
@@ -208,6 +211,7 @@ var DefaultConfig = Config{
 	},
 	CUPSJobFullUsername:          false,
 	CUPSIgnoreRawPrinters:        true,
+	CUPSIgnoreClassPrinters:      false,
```
Comment (jacobmarble) [inline lib/config.go]: Let's make this true by default.

**Comment by jacobmarble** [comment]
I looked into the raw printer filter, and found that it is condition, in manager/printermanager.go. So, raw printers are filtered twice.

Don't worry about that, though. After you fix the two comments above, we'll merge and then I'll cleanup manager/printermanager.go.

Opened #174

**Comment by tokarlss** [comment]
Thanks for the comments. I have updated the pull request accordingly.

**Comment by jacobmarble** [comment]
Thank you for your PR, Tomas.
